File: solutions/go/reverse-string/1/reverse_string.go
package reverse


func Reverse(input string) string {

    if input == "" {
        return ""
    }
    
	reversed := []rune(input)
    i := len(reversed) - 1
    for _, r := range input {
        reversed[i] = r
        i--
    }
    
    return string(reversed)
}

File: solutions/c/reverse-string/1/reverse_string.c
#include <assert.h>
#include <stdlib.h>
#include <string.h>

#include "reverse_string.h"

char* reverse(char const * value) {
    size_t length = strlen(value);
    char* reversed = (char*)malloc(length + 1);
    assert(reversed != NULL);

    char* temp = reversed;
    while (length-- > 0)
        *reversed++ = *(value + length);
    *reversed = '\0';

    return temp;
}